settings/README.md

```diff
@@ -0,0 +1,238 @@
+  let flag = false;
+
+  function loadSetting(setting: ISettingRegistry.ISettings) {
+    // Read the settings and convert to the correct type
+    limit = setting.get('limit').composite as number;
+    flag = setting.get('flag').composite as boolean;
+
+    console.log(`Limit is set to ${limit} and flag to ${flag}`);
+  }
+
+  // Wait for the application to be restored
+  app.restored
+    // Load the settings for this plugin
+    .then(() => settings.load(PLUGIN_ID))
+    .then(setting => {
+      // Read the settings
+      loadSetting(setting);
+
+      // Listen for settings changes using Signal
+      setting.changed.connect(loadSetting);
+
+      // Programmatically change a setting
+      return setting.set('limit', 20);
+    })
+    .catch(reason => {
+      console.error(
+        `Something went wrong when reading the settings.\n${reason}`
+      );
+    });
+}
+```
+<!-- prettier-ignore-end -->
+
+First, the extension waits for the application to be restored before
+loading the settings for your plugin:
+
+```ts
+// src/index.ts#L32-L32
+
+.then(() => settings.load(PLUGIN_ID))
+```
+
+Then once the settings are loaded, each setting can be read using
+the `get` method and the _setting id_ (the key defined in the settings
+JSON file). After getting the setting, you need to require the
+`composite` attribute to get its value and specify the type explicitly.
+
+```ts
+// src/index.ts#L21-L27
+
+function loadSetting(setting: ISettingRegistry.ISettings) {
+  // Read the settings and convert to the correct type
+  limit = setting.get('limit').composite as number;
+  flag = setting.get('flag').composite as boolean;
+
+  console.log(`Limit is set to ${limit} and flag to ${flag}`);
+}
+```
+
+To react at a setting change by the user, you should use the signal
+`changed`. In this case, when that signal is emitted the function
+`loadSetting` is called with the new settings.
+
+```ts
+// src/index.ts#L37-L38
+
+// Listen for settings changes using Signal
+setting.changed.connect(loadSetting);
+```
+
+Finally, in this example, a setting is changed programmatically. This
+use case can be interesting for example if some dialogs ask new values.
+
+```ts
+// src/index.ts#L40-L41
+
+// Programmatically change a setting
+return setting.set('limit', 20);
+```
+
+Once executed, the settings panel will look like this and two messages will be
+in the console:
+
+![settings example](../_images/settings_example.png)
+
+If you change the settings and refresh the page, the first message will display
+the user settings. And the second one will always have `limit` = 20.
+
+Note
+
+> This example uses [Promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise).
+> This is a technology to handle asynchronous action like reading
+> the settings in this example. So have a look at that [tutorial](https://scotch.io/tutorials/javascript-promises-for-dummies)
+> if you want to know more about the `then`/`catch` used here.
+
```

Comment:
At some point we will need a central place to link to such external resources/information to avoid duplication. Promises are going to be used in many different examples.

Reply:
That question is always delicate. People hate to click so links should be as direct as possible. Although having it at a central place will help maintenance, it will go against that.

We could also inform people in the root README of the knowledge we assume they have for understanding the examples (and points to some resource to get them). And get ride of links to non-specific phosphorjs/JupyterLab patterns in the example.

Reply:
It is very good to leave it as it is for now. As we grow, we can revisit later.